runtime/bindings/python/tests/io_runtime_test.py

```diff
@@ -0,0 +1,208 @@
+# Copyright 2023 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+import array
+import logging
+import numpy as np
+from pathlib import Path
+import tempfile
+import unittest
+
+import iree.compiler
+import iree.runtime as rt
+
+
+MM_TEST_COMPILED = None
+MM_TEST_ASM = r"""
+  #map = affine_map<(d0, d1) -> (d0, d1)>
+  #map1 = affine_map<(d0, d1) -> (d1, d0)>
+  #map2 = affine_map<(d0, d1) -> (d1)>
+  module @main {
+    util.global private @_params.classifier.weight {inlining_policy = #util.inline.never} = #stream.parameter.named<"params"::"weight"> : tensor<30x20xf32>
+    util.global private @_params.classifier.bias {inlining_policy = #util.inline.never} = #stream.parameter.named<"params"::"bias"> : tensor<30xf32>
+  func.func @run(%arg0: tensor<128x20xf32>) -> tensor<128x30xf32> {
+    %0 = call @forward(%arg0) : (tensor<128x20xf32>) -> tensor<128x30xf32>
+    return %0 : tensor<128x30xf32>
+  }
+  func.func private @forward(%arg0: tensor<128x20xf32>) -> tensor<128x30xf32> attributes {torch.assume_strict_symbolic_shapes} {
+    %cst = arith.constant 0.000000e+00 : f32
+    %_params.classifier.weight = util.global.load @_params.classifier.weight : tensor<30x20xf32>
+    %0 = tensor.empty() : tensor<20x30xf32>
+    %1 = linalg.generic {indexing_maps = [#map, #map1], iterator_types = ["parallel", "parallel"]} ins(%_params.classifier.weight : tensor<30x20xf32>) outs(%0 : tensor<20x30xf32>) {
+    ^bb0(%in: f32, %out: f32):
+      linalg.yield %in : f32
+    } -> tensor<20x30xf32>
+    %2 = tensor.empty() : tensor<128x30xf32>
+    %3 = linalg.fill ins(%cst : f32) outs(%2 : tensor<128x30xf32>) -> tensor<128x30xf32>
+    %4 = linalg.matmul ins(%arg0, %1 : tensor<128x20xf32>, tensor<20x30xf32>) outs(%3 : tensor<128x30xf32>) -> tensor<128x30xf32>
```

Comment:
it'd be nice if we had less IR in tests like this - testing that parameters work really just requires returning them from the function - no need for linalg, etc - unless the test is testing linalg or matmuls or whatnot, it shouldn't have them

(the more complex the test cases are like this the harder they become to change over time - having to come deal with #map/#map1/#map2, linalg ops, etc makes this much less amenable to cleanups/improvements)

Reply:
I'll pull this stuff out in a followup. I realized that a main reason I even wrote that was because I had forgotten to build the read side of the API and was testing it all via this integration-ey approach. I split it off to its own file and I think the non-compiler-ey tests are sufficient now, but want to take a closer look before nuking it.